### PR TITLE
Implement JSON Reads/Write for TaxableIncome model

### PR DESCRIPTION
**Pull request by narendravijayarao**

### CODE REVIEW

File: test/v2/models/TaxableIncomeSpec.scala [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+      "return a successfully read Taxable Income model" when {
+        "all fields exist" in {
+          Json.parse(validTaxableIncomeInputString).as[TaxableIncome] shouldBe validTaxableIncomeModel
+        }
+      }
+    }
+  }
+
+
+  "writes" should {
```
Comment (macgrewal) [inline test/v2/models/TaxableIncomeSpec.scala]: Add a tests to see what is rendered with the each of the optional fields is `None`

**Comment by narendravijayarao** [comment]
Just noticed... In the DES spec API24, allowancesAndDeductions is a mandatory field, but in the mapping file it is showing as Optional.
